library/Rules/Uuid.php

```diff
@@ -34,41 +37,44 @@ final class Uuid extends Standard
 {
     public const TEMPLATE_VERSION = '__version__';
 
-    private const PATTERN_FORMAT = '/^[0-9a-f]{8}-[0-9a-f]{4}-%s[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i';
-
     public function __construct(
         private readonly ?int $version = null
     ) {
+        if (!class_exists(RamseyUuid::class)) {
+            throw new MissingComposerDependencyException('Uuid rule requires ramsey/uuid', '=');
+        }
         if ($version !== null && !$this->isSupportedVersion($version)) {
             throw new InvalidRuleConstructorException(
-                'Only versions 1, 3, 4, and 5 are supported: %d given',
-                (string) $version
+                'Only versions 1 to 8 are supported: %d given',
+                (string)$version
             );
         }
     }
 
     public function evaluate(mixed $input): Result
     {
-        $template = $this->version ? self::TEMPLATE_VERSION : self::TEMPLATE_STANDARD;
+        $template   = $this->version ? self::TEMPLATE_VERSION : self::TEMPLATE_STANDARD;
         $parameters = ['version' => $this->version];
-        if (!is_string($input)) {
+
+        if (!is_string($input) && !($input instanceof UuidInterface)) {
             return Result::failed($input, $this, $parameters, $template);
         }
 
-        return new Result(preg_match($this->getPattern(), $input) > 0, $input, $this, $parameters, $template);
-    }
+        try {
+            $uuid = RamseyUuid::fromString((string) $input);
```

Comment:
```suggestion
            $uuid = $input instanceof UuidInterface ? $input : RamseyUuid::fromString($input);
```

Reply:
`$input` will either be a string or a `UuidInterface`, it's best not to convert to a string if that's already what we're expecting

Reply:
that is my mistake I was ment to remove it, well spotted